packages/firestore/test/integration/api/database.test.ts

```diff
@@ -1590,6 +1609,51 @@ apiDescribe('Database', (persistence: boolean) => {
         expect(docRef.isEqual(docRef2)).to.be.false;
       });
     });
+
+    it('Correct snapshot specified to fromFirestore() when registered with DocumentReference', () => {
+      return withTestDb(persistence, async db => {
+        const untypedDocRef = db.collection('/models').doc();
+        const docRef = untypedDocRef.withConverter(modelWithRefConverter);
+        await docRef.set(new ModelWithRef(docRef));
+        const docSnapshot = await docRef.get();
+        expect(untypedDocRef.isEqual(docSnapshot.data()!.ref)).to.be.true;
```

Comment:
This uses `refsEqual` which compares the path and the converter, but doesn't perform an instanceof check. You should add an instanceof check here and make sure that it fails before your changes.

Reply:
I changed this to:

```
const ref = docSnapshot.data()!.ref!;
expect(ref).to.be.an.instanceof(firestore.DocumentReference);
expect(untypedDocRef.isEqual(ref)).to.be.true;
```

However, at runtime it fails with that error we were talking about a few days ago:

```
Error: Cannot find module '@firebase/firestore-types'
Require stack:
- packages/firestore/test/integration/api/database.test.ts
...
```

Do you know how I can work around this problem? I'm not sure what the "correct" way is to import `DocumentReference` from `database.ts`.

Reply:
Sorry, this is a bit more complicated than I anticipated. You need to get the actual instance of DocumentReference from "firebase_export" rather than the type. "firebase_export" is used so that it can be replaced in the minified test. Here is a patch with the changes: https://gist.github.com/schmidt-sebastian/7201fe874a6f463e0ad2256780386107

Note that you must also edit ../../integration/firestore since this test file is part of the minified integration tests that runs on CI.

Reply:
Thanks for the gist! That's perfect. I've incorporated your suggestion.